server/rate_limit.go

```diff
@@ -14,10 +14,12 @@ import (
 type RateLimitTransport struct {
 	limiter *rate.Limiter
 	base    http.RoundTripper
+	metrics MetricsProvider
 }
 
 func (t *RateLimitTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	if err := t.limiter.Wait(req.Context()); err != nil {
+		t.metrics.IncreaseRateLimiterErrors()
 		return nil, err
 	}
 	return t.base.RoundTrip(req)
```

Comment:
I would also like to add a responsewriter middleware which checks for 403 responses and increments ratelimit errors in case we get a 403 from GitHub. This is because our token reserve is larger than the allowed rate limits, so in case we _think_ that we are within limits, but we make a lot of POST calls, the limiter would allow them, and trigger a rate limit error.